Plugins DataStore: decouple from onboarding

psealock opened the pull request:
Fixes https://github.com/woocommerce/woocommerce-admin/issues/3881

Make Plugins manipulation its own store to decouple it from onboarding. Selectors and resolvers surrounding installing, activating, and jetpack status or connections are now in a standalone data store in the data package.

### Differences to Fresh Data

The main difference is the inclusion of generator functions as action creators. What does this mean? Previously, analysis of `componentDidUpdate` by comparing `props` to `prevProps` was required to determine how a request was doing and what state it was in.

Now, async/await can be used to explicitly return results or errors from actions:

```js
async function install() {
	const results = await installPlugin( 'jetpack' );
	if ( results.status === 'success' ) {
		// handle success
		onInstall( results );
	} else {
		// handle error
		onError( results );
	}
}
```

### Notable Changes

* All plugin data and manipulation uses `@woocommerce/data`.
* `<Plugins />` is now an `@woocommerce/components` component.
* `<Connect />` now lives in `dashboard/components`

### Test Instructions

OBW and Task List are the only users (besides Shipping-label) to use Plugin data, so a smoke test of those two areas is the best testing.

harriswong commented:
I pulled the branch and tested the WooCommerce Service banner. 
- Tested tracking with `localStorage.setItem( 'debug', 'wc-admin:*' );`, verified events present.
- Tested no WCS installation to printing label process.
- Tested WCS installed but not activated, to printing label process
- Tested the "You've installed..." banner paragraph
![image](https://user-images.githubusercontent.com/572862/78706161-dd04e080-78cb-11ea-85e9-70fbbef2179f.png)

The WCS banner code and tests look good to me!

jeffstieler commented:
Ran into some errors while testing (unclear if all are issues with this PR or present on `master` as well):

* `An error occurred while running 'mapSelect': Cannot read property 'isActive' of undefined @ | with-plugins-hydration.js:16` (with Jetpack inactive - in console, on all page loads that use plugin data)
* `PHP Fatal error:  Uncaught Error: Class 'Automattic\WooCommerce\Admin\API\WP_Error' not found in /srv/www/wordpress-default/public_html/wp-content/plugins/woocommerce-admin/src/API/OnboardingPlugins.php:578` (when trying to click "verify business details" from WooCommerce Payments set up task

psealock commented:
Thanks for the review @jeffstieler !

>An error occurred while running 'mapSelect': ...

Weird that my setup swallowed those errors. In any case I traced it back to an assumption that Jetpack was active. Here is the fix for that https://github.com/woocommerce/woocommerce-admin/pull/4048/commits/1f9e6c204a895d2fbca95840ffa8f0ec3bd5718b

>PHP Fatal error: Uncaught Error: Class 'Automattic\WooCommerce\Admin\API\WP_Error'

Thanks for fixing up those incorrect calls to `WP_Error`. Without Jetpack, you shouldn't have gotten to that error. https://github.com/woocommerce/woocommerce-admin/pull/4105 is a recent PR to ensure that. I rebased and changed `isJetpackConnected` to come from the right store in  https://github.com/woocommerce/woocommerce-admin/pull/4048/commits/bfd0eb8f24d4b23f6e20fe6abea226f42e4813cf
